csrc/scheduler/registry.h

```diff
@@ -40,20 +41,24 @@ class TORCH_CUDA_CU_API SchedulerRuntimeInfo : public NonCopyable {
   static constexpr size_t max_alignment_size_in_byte = 16;
 
   //! Create runtime info for given fusion and input. Creating and binding
-  //!  evaluator is optional. The evaluator is used to manage intermediate
-  //!  integers in the fusion. We need them for segmenter and schedulers,
-  //!  but we don't need them when we are just using this class to provide
-  //!  additional encoding for kernel cache lookup.
+  //! evaluator is optional. The evaluator is used to manage intermediate
+  //! integers in the fusion. We need them for segmenter and schedulers,
+  //! but we don't need them when we are just using this class to provide
+  //! additional encoding for kernel cache lookup.
+  //!
+  //! The index type of forced_index_type is used if given, no matter
+  //! how large the actual arguments and fusion tensors
+  //! are. CORRECTNESS IS NOT GUARANTEED.
   SchedulerRuntimeInfo(
       Fusion* complete_fusion,
-      const KernelArgumentHolder& inputs,
-      bool create_expr_evaluator = false);
+      const KernelArgumentHolder& args,
+      PrecomputedValues* precomputed_values = nullptr,
+      const std::vector<TensorView*>& all_tvs = {},
+      std::optional<PrimDataType> forced_index_type = std::nullopt);
```

Comment:
At some point, we should change everything to `std::optional<>`, today we are mixing those two because:
1. We just bumped to C++17 not long ago, some old code using `c10::optional<>` has not been changed yet.
2. PyTorch is still using `c10::optional<>`, so for the part of the code interfacing with PyTorch, it has to be `c10::optional<>`
I don't think we have a strict rule for today about which to use. `std::optional<>` and `c10::optional<>` has the same interface, and can be easily changed in the future, so I personally just randomly pick one to use.

Reply:
Aren't we moving toward removing dependencies to the c10 and aten APIs? So, shouldn't we generally prefer `std` if possible?

Reply:
> Aren't we moving toward removing dependencies to the c10 and aten APIs? So, shouldn't we generally prefer `std` if possible?

Generally yes. But for example if the file I am modifying is already using `c10::optional` pretty heavily, I would prefer `c10::optional` for consistency.

Reply:
Yeah, if we need to cast between them, I think we should just replace everything to `std::optional` or just keep using `c10::optional` for now.